diff --git a/mk/modules.mk b/mk/modules.mk
index 6383ada4..ab0d4e89 100644
--- a/mk/modules.mk
+++ b/mk/modules.mk
@@ -86,6 +86,7 @@ USE_DTLS_SRTP := $(shell $(call CC_TEST,openssl/srtp.h))
 USE_G722      := $(shell $(call CC_TEST,spandsp/g722.h))
 USE_G722_1    := $(shell $(call CC_TEST,g722_1.h))
 USE_G726      := $(shell $(call CC_TEST,spandsp/g726.h))
+USE_G729      := $(shell $(call CC_TEST,bcg729/encoder.h))
 USE_GSM       := $(shell $(call CC_TEST,gsm.h))
 ifeq ($(USE_GSM),)
 USE_GSM       := $(shell $(call CC_TEST,gsm/gsm.h))
@@ -269,6 +270,9 @@ endif
 ifneq ($(USE_G726),)
 MODULES   += g726
 endif
+ifneq ($(USE_G729),)
+MODULES   += g729
+endif
 ifneq ($(USE_GSM),)
 MODULES   += gsm
 endif
diff --git a/src/config.c b/src/config.c
index 493dbdad..e4cd3029 100644
--- a/src/config.c
+++ b/src/config.c
@@ -937,6 +937,7 @@ int config_write_template(const char *file, const struct config *cfg)
 	(void)re_fprintf(f, "#module\t\t\t" "g7221" MOD_EXT "\n");
 	(void)re_fprintf(f, "#module\t\t\t" "g722" MOD_EXT "\n");
 	(void)re_fprintf(f, "#module\t\t\t" "g726" MOD_EXT "\n");
+	(void)re_fprintf(f, "#module\t\t\t" "g729" MOD_EXT "\n");
 	(void)re_fprintf(f, "module\t\t\t" "g711" MOD_EXT "\n");
 	(void)re_fprintf(f, "#module\t\t\t" "gsm" MOD_EXT "\n");
 	(void)re_fprintf(f, "#module\t\t\t" "l16" MOD_EXT "\n");
